core/state/pruner/pruner.go

```diff
@@ -228,10 +233,130 @@ func prune(snaptree *snapshot.Tree, root common.Hash, maindb ethdb.Database, sta
 	return nil
 }
 
+func nodeIteratorKey(it trie.NodeIterator) (common.Hash, error) {
+	if it.Leaf() {
```

Comment:
Looking through extractGenesis it doesn't have this part - only it.Hash()
Also - if I'm reading this correctly, LeafKey would be the path to the data, and I don't remember that it should appear as a key in the database (?)

Reply:
extractGenesis is right here :) I'm updating my function and having extractGenesis just call mine since mine is parallel and has ETA tracking